consensus/consensus.go

```diff
@@ -55,4 +55,5 @@ type Factory func(
 	*state.Executor,
 	*grpc.Server,
 	hclog.Logger,
+	*Metrics,
```

Comment:
Why can't we have the metrics field be part of the consensus Config?

Reply:
I felt like having metrics in Config will make code more clumsy, since fields of metrics type will be used frequently to update the counters across the flow.